PROD4POD-1830 Add Onboarding screen

mteresa-jimenez opened the pull request:
# ✍️ Description
Add `Onboarding` screen:
- Implement font and basic styles for the feature.
- Implement screen structure.
- Add translation strings in `locales` (content is still TBD).
- Create new FixedFooter component in poly-look.

### 🏗️ Fixes [PROD4POD-1830](https://jira.polypoly.eu/browse/PROD4POD-1830)

## ℹ️ Other information
[Figma link](https://www.figma.com/file/H4JnISIsC06WPouox7CITh/MAIN-pod-GDPR-helper-feature?node-id=1698%3A11153)

Buttons navigation will be added after Router is implemented in the feature.

JJ commented on feature-utils/poly-look/src/react-components/fixedFooter.jsx:
```diff
@@ -0,0 +1,9 @@
+import "./fixedFooter.css";
+
+const FixedFooter = ({ children }) => {
+  return <div className="fixed-footer">{children}</div>;
```
Since this seems to be essentially a div with a certain class name, would there be a way of having a generic component that you can instantiate with class names? I lost track of all the components, but is there a chance you could use the same pattern elsewhere?

mteresa-jimenez replied:
What do you mean by a generic component? Some `Container` component with a `div` and a list of classNames that we could chose?
 
This is something I asked @fhd last week, because I wasn't sure if adding a component or just a className in poly-look. He agreed in the idea of creating a new component (perhaps he changed his mind and thinks the generic component is a better idea, though 😅 ).

Another option is organising better the components that we have and group them in different folders.

JJ replied:
> What do you mean by a generic component? Some `Container` component with a `div` and a list of classNames that we could chose?

Yea, pretty much. Even without a closed list of class names. You chose one with CSS, you're good.
> 
> This is something I asked @fhd last week, because I wasn't sure if adding a component or just a className in poly-look. He agreed in the idea of creating a new component (perhaps he changed his mind and thinks the generic component is a better idea, though sweat_smile ).
> 

It was a suggestion; but it makes sense and will reduce the amount of boilerplate if you use this pattern repeatedly.

> Another option is organising better the components that we have and group them in different folders.

That will certainly help find duplications and DRYing, but it's a separate issue.

g3o314 replied:
I was wondering about the css class vs component as well since this is such a simple component, at least now I know 😂 
As for the generic `Container` I'm not so sure. We get rid of some duplication but we also loose the meaning/intention behind the component. In this case, we might as well get rid of the component(s) entirely and stick to css classes 🤷

mteresa-jimenez replied:
I have added the `gradient` as a boolean option to make the component a bit more versatile - for example, for the "Log out" that some other screens will have. It's still a simple component, but at least now it's not just a `div` and could be more reusable.

JJ reviewed (approved):
You probably need to make a symbolic link to make them available, but the objection is removed. Thanks!

mteresa-jimenez commented:
> You probably need to make a symbolic link to make them available, but the objection is removed. Thanks!

I was going to say if you could check this commit [6ce382f](https://github.com/polypoly-eu/polyPod/pull/1172/commits/6ce382fdb84bd87e468ccb92cb78a55cd609aad1)

I'm not sure if I need to add that to the rollup, but the fonts work without it.

JJ commented:
I've restarted the Android integration test, just in case.

g3o314 commented on feature-utils/poly-look/src/react-components/fixedFooter.jsx:
```diff
@@ -0,0 +1,9 @@
+
+import "./fixedFooter.css";
+
+const FixedFooter = ({ children }) => {
```
Please add documentation.

g3o314 commented:
Because the footer component is pretty straightforward, I would like to see it added to Storybook. We can pair on that if you want.
It would be nice if we could test it, but I have no idea how to setup the proper test case. For this particular component I would be more interested in testing that it actually stays fixed as opposed to "renders the children".

g3o314 reviewed (approved):
Well, I think the fonts are the only thing missing, but because the build doesn't break and since we already have a ticket for the fonts, I think we are good to go.

mteresa-jimenez commented:
> Because the footer component is pretty straightforward, I would like to see it added to Storybook. We can pair on that if you want. It would be nice if we could test it, but I have no idea how to setup the proper test case. For this particular component I would be more interested in testing that it actually stays fixed as opposed to "renders the children".

@g3o314 I have added the component to Storybook. If you move the "addons menu" up and down, you will see that the component is fixed to the bottom. The gradient option is trickier to see here, but if you select "side by side" option in the Themes, you can appreciate it.

Regarding the tests, I couldn't find a way to _exactly_ test the CSS styles if they are in a separated CSS file - Jest didn't "detect" the styles in from the CSS class, but ir did if I added something like `styles={{position: "fixed}}` to the `div`. So I have tested that the component has the `poly-fixed-footer` class.

mteresa-jimenez commented:
> Well, I think the fonts are the only thing missing, but because the build doesn't break and since we already have a ticket for the fonts, I think we are good to go.

Yes, I've seen that there was a separated ticket for the fonts and didn't want to complicate more this PR. 

Thank you!
